### PR TITLE
Fetch OSM Points of Interest (resolves #116)

### LINKED ISSUE #116: New feature: Retrieving OSM POIs with osmnx?
Hi, 

first of all, awesome work with the package! Highly useful and needed. 

I was wondering if there would be interest to include separate functionalities to retrieve OpenStreetMap POIs with osmnx? I am currently starting to write such functionalities for my own project, and I think I will be using osmnx as a basis for fetching Points of interest from OpenStreetMap (such as restaurants etc.). I know that it is possible to work with [other infrastructures](https://github.com/gboeing/osmnx-examples/blob/master/notebooks/16-work-with-other-infrastructures.ipynb) than roads, but if I have understood correctly that function works only with graphs, not with point data such as POIs. 

If you think this would be interesting, I would be happy to help implementing these functions. 

Cheers,
Henrikki

P.S. FYI: I introduce osmnx also with the fully open GIS course (Lesson 7) that can be found here [http://autogis.github.io](http://autogis.github.io).

### PR DESCRIPTION
**Pull request by HTenkanen**
Implemented functionalities to fetch points of interests from OpenStreetMap using osmnx that can be found from `pois.py` file. Tests were written in `tests/test_osmnx.py`. All functions include docstrings and are written following the same style as other osmnx functionalities. 

The implemented functions in `pois.py` are:
 - def pois_from_place(place, amenities=None)
 - def pois_from_point(point, distance=None, amenities=None)
 - def pois_from_address(address, distance, amenities=None)
 - def pois_from_polygon(polygon, amenities=None)

`amenities` parameter allows to filter specific types of OSM amenities such as `restaurant`, `school` etc.

### CODE REVIEW

**Comment by gboeing** [comment]
resolves #116

**Comment by jimr1603** [comment]
I really need to read more of GitHub before diving into stuff. I've been doing something similar with osmosis for pre-filtering. This is a great feature

File: osmnx/utils.py [hunk 1]
```diff
@@ -997,4 +995,9 @@ def overpass_json_from_file(filename):
         handler = OSMContentHandler()
         xml.sax.parse(file, handler)
 
+        #import pprint; pprint.pprint(handler.object)
```
Comment (gboeing) [inline osmnx/utils.py]: This commented-out line can be removed.

File: osmnx/pois.py [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+# License: MIT, see full license in LICENSE.txt
+# Web: https://github.com/gboeing/osmnx
+################################################################################
+
+from shapely.geometry import Point, LineString, MultiPolygon, Polygon, box
+import geopandas as gpd
+from .core import overpass_request, bbox_from_point, gdf_from_place
+from .utils import log, geocode, bbox_to_poly
+
+def parse_poi_query(west, south, east, north, amenities=None, timeout=180, maxsize=''):
```
Comment (gboeing) [inline osmnx/pois.py]: This is probably the ideal argument order, but everywhere else in OSMnx, bounding-box extents are passed as north, south, east, west. For the sake of consistency, could you tweak the argument order as such?

File: osmnx/pois.py [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+        Northernmost coordinate from bounding box of the search area.
+    amenities : list
+        List of amenities that will be used for finding the POIs from the selected area.
+    timeout : int
+        Timeout for the API request. 
+    """
+    if amenities:
+        # Overpass QL template
+        query_template = ('[out:json][timeout:{timeout}]{maxsize};((node["amenity"~"{amenities}"]({south:.8f},'
+                          '{west:.8f},{north:.8f},{east:.8f});(._;>;););(way["amenity"~"{amenities}"]({south:.8f},'
```
Comment (gboeing) [inline osmnx/pois.py]: For consistency, could you round all these coordinates (and below) to 6 decimal places?

File: osmnx/pois.py [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+    Parses the osm relations from osm ways and nodes.
+         
+    Parameters
+    ==========
+    relations : list
+        OSM 'relation' items (dictionaries) in a list. 
+    osm_way_df : gpd.GeoDataFrame
+        OSM 'way' features as a GeoDataFrame that contains all the 'way' features that will constitute the multipolygon relations.
+     
+    Information
```
Comment (gboeing) [inline osmnx/pois.py]: Can we remove the "information" header and move the URL into the top of the docstring with the sentence description?

File: osmnx/utils.py (hunk 1 above)
Comment (HTenkanen) [reply osmnx/utils.py]: Done.

File: osmnx/pois.py (hunk 2 above)
Comment (HTenkanen) [reply osmnx/pois.py]: Done.

File: osmnx/pois.py (hunk 3 above)
Comment (HTenkanen) [reply osmnx/pois.py]: Done.

File: osmnx/pois.py (hunk 4 above)
Comment (HTenkanen) [reply osmnx/pois.py]: Yes a good idea. Done.

**Comment by HTenkanen** [comment]
@gboeing Done now with the minor edits according your comments! 👍 Very happy to contribute, this is a great package! 🙂

**Pull request merged**